jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/Jsonschema2Pojo.java

```diff
@@ -127,6 +127,10 @@ private static void generateRecursive(GenerationConfig config, SchemaMapper mapp
 
         for (File child : schemaFiles) {
             if (child.isFile()) {
+                if (config.getSourceType() == SourceType.JSON || config.getSourceType() == SourceType.YAML) {
+                    // any cached schemas will have ids that are fragments, relative to the previous document (and shouldn't be reused)
+                    mapper.getRuleFactory().getSchemaStore().clearCache();
```

Comment:
Before implementing https://github.com/joelittlejohn/jsonschema2pojo/pull/1464, I have also thought about clearing cache but for some reason it felt somewhat weird to have source type checks.
Now that there is a nice comment it does not feel so "alien" anymore 👍

Reply:
Yes, I must admit, it feels like a bit of a hack.

I implemented a couple of alternatives before landing on the solution you see in this PR. First I thought lets put the JSON filename into the ID, however this means that we will try to resolve content from it (and it led me down the same path as you). Next I thought let's not cache the schemas at all in this case (why bother?) but then found that this breaks recursive references since you get into an infinite loop of resolving schemas if you don't remember (and reuse) what you have resolved in the past (yay for extensive integration tests!).

After these two failures, I decided that if we are building a cache of schemas that must refer only to one document, because all references are root-relative, then just clear them out between files. Hopefully the comment stops the reasoning being completely lost to the mists of time.

It could be better to 'qualify' the schemas in the cache in a better way, or to unify everything as you did in your PR, however I like how little additional change and conceptual complexity my change brings in, so it felt like the best of all available solutions.

Reply:
Given approach indeed is far simpler than unifying one.
Even though latter would have made `json` and `yaml` source types "fisrt class citizens", clearing cache makes sense as well.
Thank you for providing "insights" 👍